lib/beaker/hypervisor/abs.rb

```diff
@@ -13,6 +13,23 @@ def initialize(hosts, options)
       @resource_hosts = JSON.parse(resource_hosts)
     end
 
+    # Set ssh connection method preference for each host
+    #
+    # The default order that is already set is ['ip', 'vmhostname', 'hostname']
+    # If an engine is using default order, return nothing.
+    # Specify this for each engine or it will raise an error.
+    # If you are changing this, make sure the elements' order is changed
+    # and not the elements themselves.
+    def get_ssh_connection_preference(engine)
+      case engine
+      when /^(vmpooler|nspooler)$/
+        # putting ip last as its not set by ABS
+        return ['vmhostname', 'hostname', 'ip']
+      else
+        return ['vmhostname', 'hostname', 'ip']
```

Comment:
I was raising an error as the default case instead of returning this array. @nwolfe suggested to return the array with `vmhostname` as the first element as then it would not hinder ABS development (addition of new engines) as ABS always returns DNS name (`vmhostname`) and it would go to the default case.

The point of whole switch case is to make this forward compatible, if any engine(in future) returns hosts that you can SSH with IP only then you can add a case for that engine and make IP as your first preference in the array.